sisl/viz/plots/grid.py

```diff
@@ -1525,17 +1530,22 @@ def _set_data(self, i, geometry, grid, k, grid_prec, nsc):
         if self.geometry is None:
             raise ValueError('No geometry was provided and we need it the basis orbitals to build the wavefunctions from the coefficients!')
 
+        # Get the spin class for which the eigenstate was calculated.
+        spin = sisl.Spin()
+        if self.eigenstate.parent is not None:
```

Comment:
if `eigenstate.parent is None` then basically nothing can be used here?

Reply:
Hmm do you mean because there is no geometry? The geometry is determined in the lines above. It can be passed explicitly as a setting (the `geometry` setting), just in case we could not automatically find it. After going through the multiple ways the geometry can be retrieved, if there was still no geometry, an error is raised (line 1531). 

So at this point we already know we have a geometry, we are just trying to see if we know what the spin class of the calculation was.

Reply:
Yeah ok, I would just try never to use a `geometry` as a parent in the eigenstate approach, *only* for wavefunctions this would be ok.

Reply:
For bands it is ok. :)